Extension/src/LanguageServer/client.ts

```diff
@@ -1842,7 +1842,8 @@ export class DefaultClient implements Client {
     public onDidChangeTextDocument(textDocumentChangeEvent: vscode.TextDocumentChangeEvent): void {
         if (util.isCpp(textDocumentChangeEvent.document)) {
             // If any file has changed, we need to abort the current rename operation
-            if (workspaceReferences.renamePending) {
+            if (workspaceReferences !== undefined // Occurs when a document changes before cpptools starts.
```

Comment:
It looks like `onDidChangeTextDocument` is registered in `activate` (in `extension.ts`), after clients have been created and `init` called. But `init` is async and performs an await before setting `workspaceReferences`.

Could you maybe add an `await this.ready;` instead?  It's possible all of the handler installed (early) in `activate` should also `await this.ready;`, unless there is some reason they should run before `init` is complete.

Reply:
ok

Reply:
But I'm not sure what other handlers should have it added too -- they all seem okay to me.